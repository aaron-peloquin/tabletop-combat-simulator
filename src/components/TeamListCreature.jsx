import {
  Button,
  Card,
  CardActions,
  CardContent,
  Typography,
} from "@material-ui/core/"
// import { withStyles } from "@material-ui/core/styles"
import { connect } from "react-redux"

import SetEditCreature from "./../store/action/SetEditCreature"
import DeleteCreature from "./../store/action/DeleteCreature"

const TeamListCreature = (props) => {
  const {
    /** Actions */
    FuncSetEditCreature,
    FuncDeleteCreature,

    /** Attributes */
    Creature,
  } = props
  return <Card>
    <CardContent>
      <Typography variant="body2">{Creature.name}</Typography>
    </CardContent>
    <CardActions>
      <Button onClick={() => { FuncSetEditCreature(Creature) }}>Edit</Button>
      <Button onClick={() => { FuncDeleteCreature(Creature.hash) }}>Delete</Button>
    </CardActions>
  </Card>
}

const MapStateToProps = () => {
  return {}
}

const MapActionsToProps = (dispatch) => {
  console.log("dispatch", typeof dispatch)
  return {
    FuncSetEditCreature: (Creature) => {
      console.log("SetEditCreature", typeof dispatch)
      SetEditCreature(dispatch, Creature)
    },
    FuncDeleteCreature: (Hash) => {
      DeleteCreature(dispatch, Hash)
    },
  } 
}

export default connect(MapStateToProps, MapActionsToProps)(TeamListCreature)
