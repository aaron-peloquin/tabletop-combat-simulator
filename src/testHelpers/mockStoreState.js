export default {
  "creatures": [
    {
      "name": "Goplin",
      "cr": "1/4",
      "hash": "hivn3p0k",
      "description": "A stout monster",
      "defaultInitiative": 10,
      "armor": 11,
      "hp": 5,
      "hitDiceEquation": "1d20+1",
      "damageDiceEquation": "1d4"
    },
    {
      "name": "Elven Ranger",
      "cr": "1/2",
      "hash": "h3dxf5wl",
      "description": "A swift hunter",
      "defaultInitiative": 16,
      "armor": 13,
      "hp": 10,
      "hitDiceEquation": "1d20+8",
      "damageDiceEquation": "1d6+3"
    },
    {
      "name": "Knight",
      "cr": "1",
      "hash": "hee9p286",
      "description": "An honorable soldier",
      "defaultInitiative": 11,
      "armor": 18,
      "hp": 25,
      "hitDiceEquation": "1d20+4",
      "damageDiceEquation": "1d12+4"
    },
    {
      "name": "Lizardfolk",
      "cr": "3",
      "hash": "howx26rt",
      "description": "An ancient sneaky monster",
      "defaultInitiative": 14,
      "armor": 14,
      "hp": 15,
      "hitDiceEquation": "1d20+5",
      "damageDiceEquation": "1d10+2"
    }
  ],
  "sideBar": false,
}