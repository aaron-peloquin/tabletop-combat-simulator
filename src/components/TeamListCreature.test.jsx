import toJson from "enzyme-to-json"
import {shallow} from "enzyme"

import {RawTeamListCreature} from "./TeamListCreature"
import mockStoreState from "../testHelpers/mockStoreState"

jest.mock("@material-ui/core/")

describe("<TeamListCreature />", () => {
  // const renderShallowUntilComponent = createShallow({"untilSelector": "TeamListCreature"})
  const props = {
    Creature: mockStoreState.creatures[0],
    funcSetEditCreature: jest.fn(()=>{}),
    funcCopyCreature: jest.fn(()=>{}),
    funcDeleteCreature: jest.fn(()=>{}),
    classes: {CreatureCard: true},
  }
  const TeamListCreatureComponent = shallow(<RawTeamListCreature {...props} />)

  it("renders", () => {
    expect(typeof TeamListCreatureComponent).toBe("object")
  })

  it("calls edit function", () => {
    const editButton = TeamListCreatureComponent.find("[data-testid='editButton']")
    expect(editButton.length).toBe(1)
    editButton.simulate("click")
    expect(props.funcSetEditCreature).toHaveBeenCalledTimes(1)
  })

  it("calls copy function", () => {
    const copyButton = TeamListCreatureComponent.find("[data-testid='copyButton']")
    expect(copyButton.length).toBe(1)
    copyButton.simulate("click")
    expect(props.funcCopyCreature).toHaveBeenCalledTimes(1)
  })

  it("calls delete function", () => {
    const deleteButton = TeamListCreatureComponent.find("[data-testid='deleteButton']")
    expect(deleteButton.length).toBe(1)
    deleteButton.simulate("click")
    expect(props.funcDeleteCreature).toHaveBeenCalledTimes(1)
  })

  it("snapshots", () => {
    const Tree = toJson(TeamListCreatureComponent)
    expect(Tree).toMatchSnapshot()
  })
})
