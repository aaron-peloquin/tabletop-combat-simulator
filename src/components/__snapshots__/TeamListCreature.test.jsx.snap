// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamListCreature /> snapshots 1`] = `
<WithStyles(Card)
  className={true}
>
  <WithStyles(CardContent)>
    <WithStyles(Grid)
      alignItems="stretch"
      container={true}
      justify="center"
      spacing={8}
    >
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <WithStyles(Typography)
          variant="body1"
        >
          <strong>
            Goplin
          </strong>
        </WithStyles(Typography)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={4}
      >
        <WithStyles(Typography)
          variant="body2"
        >
          <WithStyles(Component) />
           
          5
        </WithStyles(Typography)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={4}
      >
        <WithStyles(Typography)
          variant="body2"
        >
          <WithStyles(Component) />
           
          11
        </WithStyles(Typography)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={4}
      >
        <WithStyles(Typography)
          variant="body2"
        >
          <WithStyles(Component) />
           +
          10
        </WithStyles(Typography)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={6}
      >
        <WithStyles(Typography)
          variant="body2"
        >
          <WithStyles(Component) />
           
          1d20+1
        </WithStyles(Typography)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={6}
      >
        <WithStyles(Typography)
          variant="body2"
        >
          <WithStyles(Component) />
           
          1d4
        </WithStyles(Typography)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(CardContent)>
  <WithStyles(CardActions)>
    <WithStyles(Grid)
      alignItems="stretch"
      container={true}
      justify="center"
      spacing={8}
    >
      <WithStyles(Grid)
        item={true}
        xs={4}
      >
        <WithStyles(Button)
          color="primary"
          data-testid="editButton"
          onClick={[Function]}
          variant="contained"
        >
          <pure(EditIcon) />
        </WithStyles(Button)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={4}
      >
        <WithStyles(Button)
          color="primary"
          data-testid="copyButton"
          onClick={[Function]}
          variant="contained"
        >
          <pure(FileCopyIcon) />
        </WithStyles(Button)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={4}
      >
        <WithStyles(Button)
          color="secondary"
          data-testid="deleteButton"
          onClick={[Function]}
          variant="contained"
        >
          <pure(DeleteIcon) />
        </WithStyles(Button)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(CardActions)>
</WithStyles(Card)>
`;
